src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -901,25 +901,23 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
           // "Capture" the error by finding the nearest boundary. If there is no
           // error boundary, the nearest host container acts as one. If
           // captureError returns null, the error was intentionally ignored.
-          const maybeBoundary = captureError(failedWork, error);
-          if (maybeBoundary !== null) {
-            const boundary = maybeBoundary;
-
-            // Complete the boundary as if it rendered null. This will unmount
-            // the failed tree.
-            beginFailedWork(boundary.alternate, boundary, nextPriorityLevel);
-
-            // The next unit of work is now the boundary that captured the error.
-            // Conceptually, we're unwinding the stack. We need to unwind the
-            // context stack, too, from the failed work to the boundary that
-            // captured the error.
-            // TODO: If we set the memoized props in beginWork instead of
-            // completeWork, rather than unwind the stack, we can just restart
-            // from the root. Can't do that until then because without memoized
-            // props, the nodes higher up in the tree will rerender unnecessarily.
-            unwindContexts(failedWork, boundary);
-            nextUnitOfWork = completeUnitOfWork(boundary);
-          }
+          const boundary = captureError(failedWork, error);
+          invariant(
+            boundary !== null,
+            'Should have found an error boundary. This error is likely ' +
+              'caused by a bug in React. Please file an issue.',
+          );
+          // The next unit of work is now the boundary that captured the error.
+          // Conceptually, we're unwinding the stack. We need to unwind the
+          // context stack, too, from the failed work to the boundary that
+          // captured the error.
+          unwindContexts(failedWork, boundary);
```

Comment:
Turns out this isn't observable because `beginFailedWork` doesn't read from context. All it does is render null. So the order of `beginFailedWork` and `unwindContext` ends up not mattering. I'll improve the test anyway, but that's why we weren't getting errors before.